py/jpy/src/main/c/jpy_jtype.c

```diff
@@ -861,7 +895,9 @@ int JType_ConvertPythonToJavaObject(JNIEnv* jenv, JPy_JType* type, PyObject* pyA
         return JPy_AsJString(jenv, pyArg, objectRef);
     } else if (PyBool_Check(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Boolean_JClass, type->classRef)))) {
         return JType_CreateJavaBooleanObject(jenv, type, pyArg, objectRef);
-    } else if (JPy_IS_CLONG(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Integer_JClass, type->classRef)))) {
+    } else if (JPy_IS_CLONG(pyArg) && (type == JPy_JObject )) {
+        return JType_CreateJavaIntegerTypeObjectByInference(jenv, type, pyArg, objectRef);
```

Comment:
I think `JavaIntegerType` is still not the best name. I *think* something like:

```c
    } else if (JPy_IS_CLONG(pyArg) && ((*jenv)->IsAssignableFrom(jenv, JPy_Number_JClass, type->classRef))) {
        return JType_CreateJavaNumberObject(jenv, type, pyArg, objectRef);
```

Not sure if `JPy_Number_JClass` exists, but would point to `java.lang.Number`.